packages/chain-bridge/src/adapters/StargateAdapter.sol

```diff
@@ -818,6 +1143,137 @@ contract StargateAdapter is Ownable, IBridgeAdapter, ILayerZeroComposer {
         );
     }
 
+    /**
+     * @dev Try to deposit assets to FleetCommander without reverting on failure
+     */
+    function _tryDepositToFleetCommander(
+        address fleetCommander,
+        address asset,
+        uint256 amount,
+        address shareRecipient,
+        bytes memory referralCode,
+        bytes32 operationId,
+        address, // originalUser - not used in current implementation
+        address, // sourceAsset - not used in current implementation
+        uint16 sourceChainId
+    ) internal returns (bool success) {
+        // Validate FleetCommander supports the asset
+        try IFleetCommanderMinimal(fleetCommander).asset() returns (
+            address fleetAsset
+        ) {
+            if (fleetAsset != asset) {
+                emit CrossChainFleetDepositFailed(
+                    operationId,
+                    fleetCommander,
+                    asset,
+                    amount,
+                    "Asset mismatch"
+                );
+                return false;
+            }
+        } catch {
+            emit CrossChainFleetDepositFailed(
+                operationId,
+                fleetCommander,
+                asset,
+                amount,
+                "Asset check failed"
+            );
+            return false;
+        }
+
+        // Check deposit limits
+        try
+            IFleetCommanderMinimal(fleetCommander).maxDeposit(address(this))
```

Comment:
assuming we have the validated fleet commander ( from harbor command) no need for the try catch.

and revert  instead of the event. 

Havent gone through all the contracts yet - but not sure if any state changes are required.